al_servicec.js

```diff
@@ -259,10 +260,35 @@ class AzcollectC extends AlServiceC {
         };
         const type = this._collectorType;
         var functionName = encodeURIComponent(checkinValues.functionName);
-        return this.post(`/aws/${type}/checkin/` +
-            `${checkinValues.awsAccountId}/${checkinValues.region}/${functionName}`, {body: statusBody});
+        var checkinUrl = `/aws/${type}/checkin/${checkinValues.awsAccountId}/` +
+                         `${checkinValues.region}/${functionName}`;
+        var sendCompressed = process.env.send_checking_compressed ?
+                             process.env.send_checking_compressed : true;
+        if (sendCompressed) {
+            zlib.deflate(JSON.stringify(statusBody), function(err, data) {
+                if (err) {
+                    return new Promise(function(resolve, reject) {
+                        return reject(err);
+                    });
+                } else {
+                    let payload = {
+                        headers : {
+                            'Content-Encoding' : 'deflate',
+                            'Content-Length' : Buffer.byteLength(data)
+                        },
+                        body : data
+                    };
+                    return this.post(checkinUrl, payload);
```

Comment:
@kkuzmin tests are currently failing:

```
 Suite duration: 0 s, Tests: 0

  3 passing (21ms)
  2 failing

  1) Unit Tests AzcollectC checkin:
     TypeError: Cannot read property 'then' of undefined
      at Context.<anonymous> (test/azcollectc_test.js:125:39)

  2) Unit Tests AzcollectC "after each" hook:
     Uncaught TypeError: Cannot read property 'post' of undefined
```

could you please help me fixing that?